tests/backup/backup_test.go

```diff
@@ -1011,467 +1025,765 @@ var _ = Describe("{BackupCrashVolDriver}", func() {
-		Bucket: aws.String(bucketName),
-	})
-	Expect(err).NotTo(HaveOccurred(),
-		fmt.Sprintf("Failed to create bucket [%v]. Error: [%v]", bucketName, err))
+			ValidateApplications(contexts)
+			for _, ctx := range contexts {
+				for vol, params := range GetVolumeParameters(ctx) {
+					volumeParams[vol] = params
+				}
+			}
+		})
 
-	err = S3Client.WaitUntilBucketExists(&s3.HeadBucketInput{
-		Bucket: aws.String(bucketName),
-	})
-	Expect(err).NotTo(HaveOccurred(),
-		fmt.Sprintf("Failed to wait for bucket [%v] to get created. Error: [%v]", bucketName, err))
-}
+		// TODO(stgleb): Add multi-namespace backup when ready in px-backup
+		for _, namespace := range bkpNamespaces {
+			go func(namespace string) {
+				backupName := fmt.Sprintf("%s-%s", backupNamePrefix, namespace)
+				Step(fmt.Sprintf("Create backup full name %s:%s:%s",
+					sourceClusterName, namespace, backupName), func() {
+					err = CreateBackupGetErr(backupName,
+						sourceClusterName, backupLocationName, backupLocationUID,
+						[]string{namespace}, labelSelectors, orgID)
+					if err != nil {
+						bkpNamespaceErrors[namespace] = err
+					}
+				})
+			}(namespace)
+		}
 
-func CreateAzureBucket(bucketName string) {
-	// From the Azure portal, get your Storage account blob service URL endpoint.
-	_, _, _, _, accountName, accountKey := getAzureCredsFromEnv()
+		var wg sync.WaitGroup
+		for _, namespace := range bkpNamespaces {
+			backupName := fmt.Sprintf("%s-%s", backupNamePrefix, namespace)
+			error, ok := bkpNamespaceErrors[namespace]
+			if ok {
+				logrus.Warningf("Skipping waiting for backup %s because %s", backupName, error)
+			} else {
+				wg.Add(1)
+				go func(wg *sync.WaitGroup, namespace, backupName string) {
+					defer wg.Done()
+					Step(fmt.Sprintf("Wait for backup %s to complete", backupName), func() {
+
+						ctx, err := backup.GetPxCentralAdminCtx()
+						Expect(err).NotTo(HaveOccurred(),
+							fmt.Sprintf("Failed to fetch px-central-admin ctx: [%v]",
+								err))
+						err = Inst().Backup.WaitForBackupCompletion(
+							ctx,
+							backupName, orgID,
+							backupRestoreCompletionTimeoutMin*time.Minute,
+							retrySeconds*time.Second)
+						if err != nil {
+							bkpNamespaceErrors[namespace] = err
+							logrus.Errorf("Failed to wait for backup [%s] to complete. Error: [%v]",
+								backupName, err)
+						}
+					})
+				}(&wg, namespace, backupName)
+			}
+		}
+		wg.Wait()
 
-	urlStr := fmt.Sprintf("https://%s.blob.core.windows.net/%s", accountName, bucketName)
-	logrus.Infof("Create container url %s", urlStr)
-	// Create a ContainerURL object that wraps a soon-to-be-created container's URL and a default pipeline.
-	u, _ := url.Parse(urlStr)
-	credential, err := azblob.NewSharedKeyCredential(accountName, accountKey)
-	Expect(err).NotTo(HaveOccurred(),
-		fmt.Sprintf("Failed to create shared key credential [%v]", err))
+		successfulBackups = len(bkpNamespaces) - len(bkpNamespaceErrors)
```

Comment:
It would be good if we can capture, which backups failed and what the corresponding errors were.